ddp-workspace/projects/ddp-sdk/src/lib/services/authentication/auth0Adapter.service.ts

```diff
@@ -201,34 +214,31 @@ export class Auth0AdapterService implements OnDestroy {
         });
     }
 
-    public auth0RenewToken(): void {
-        const participantGuid = this.session.session.participantGuid;
-        this.session.sessionObservable.pipe(
-            // Because the session is BehaviorSubject, we should skip a current session,
-            // we are only interested in the renewed session
-            skip(1),
-            take(1)
-        ).subscribe(() => {
-            this.renewNotifier.hideSessionExpirationNotifications();
-        });
-        // Original code called renewAuth. This seems to be renewing token correctly
-        // Note the response types. 'code' is explicitly not supported
-        this.webAuth.checkSession(
-            {
-                studyGuid: this.configuration.studyGuid,
-                responseType: 'token id_token'
-            },
-            (err, result) => {
-                if (err) {
-                    console.log(err);
-                } else {
-                    console.log('auth0Adapter.renewToken success', result);
-                    this.setSession({
-                        ...result,
-                        participantGuid
-                    });
-                }
-            });
+    public auth0RenewToken(): Observable<Session | null>  {
+        const studyGuid = this.configuration.studyGuid;
+        const auth0IdToken = this.jwtHelper.decodeToken(this.session.session.idToken)['sub'];
+        const clientId = this.configuration.auth0ClientId;
+        const resultMatchesThisSession = (result: any) => {
+            const resultClientId: any = result.idTokenPayload['https://datadonationplatform.org/cid'];
+            return result.idTokenPayload['sub'] === auth0IdToken && resultClientId && resultClientId === clientId;
+        };
+        const checkSession$ = bindNodeCallback(cb => this.webAuth.checkSession({
+            studyGuid,
+            responseType: 'token id_token',
+            renew_token_only: true // this flag will indicate that Auth0 should not try to call user registration
+        }, cb));
+        return checkSession$().pipe(
+                take(1),
+                tap(result => {
+                    if (resultMatchesThisSession(result)) {
+                        this.renewSession(result);
+                    } else {
+                        throw new Error('Token received does not match this session');
+                    }
+                }),
+                tap(result => this.renewNotifier.hideSessionExpirationNotifications()),
+                mergeMap(result => this.session.sessionObservable.pipe(take(1)))
+        );
```

Comment:
Please, format the code here, indents look weird.

Reply:
3ac3e5b